### DISCUSSION TITLE
Using it with unity

### DISCUSSION THREAD

**Discussion by ezzatmoussa** [FantasticFiasco/aws-signature-version-4#474] [Q&A] [answered]
Have anyone tried using it with unity?
I have an issue in il2cpp related to System.Web dll 

`Error: called non-existent method System.Collections.Specialized.NameValueCollection System.Web.HttpUtility::ParseQueryString(System.String)
`

**Comment by FantasticFiasco** [comment]
Is it Unity as in the cross-platform game engine?

**Reply by ezzatmoussa** [reply]
Yes exactly, the problem is that it doesn't fully support System.Web dll

**Reply by FantasticFiasco** [reply]
Yikes, I have absolutely no experience on that platform. I'll leave the discussion here and if you or any other developer finds the solution, please describe it here. If the solution requires and changes to this repo, please let me know and we'll see what we can do.

Good luck!

**Reply by ezzatmoussa** [reply]
Thank you, I will check the repo and see what options are available.

**Comment by ezzatmoussa** [comment, marked as answer]
It appears that the package would normally work in Unity with someone using .net standard 2. but since I have packages that require .net 4.5 
Where System.Web isn't available; I built the library after replacing System.Web ParseQueryString by helper scripts found in [Nancy repo](https://github.com/NancyFx/Nancy),  and now it works; it appears the only issue was System.Web dll.

**Reply by FantasticFiasco** [reply]
Would you benefit from me removing the dependency to `System.Web ParseQueryString`? Would you be able to use the NuGet package as is then? If yes, I would happily create a branch and build you a beta to try.

**Reply by ezzatmoussa** [reply]
Yeah, that would be great, and it should work as it is the only dependency not supported by Unity.

**Reply by FantasticFiasco** [reply]
v2.1.0-beta.1 is now [live on nuget.org](https://www.nuget.org/packages/AwsSignatureVersion4/2.1.0-beta.1). Could you please verify that it works in your environment?

**Reply by ezzatmoussa** [reply]
It works fine now. Many thanks.

**Reply by FantasticFiasco** [reply]
v2.1.0 has now been released. Thanks for reporting the issue!
